cpp/src/arrow/scalar.h

```diff
@@ -345,8 +345,8 @@ struct ARROW_EXPORT TimestampScalar : public TemporalScalar<TimestampType> {
   using TemporalScalar<TimestampType>::TemporalScalar;
 
   TimestampScalar(typename TemporalScalar<TimestampType>::ValueType value,
-                  TimeUnit::type unit)
-      : TimestampScalar(std::move(value), timestamp(unit)) {}
+                  TimeUnit::type unit, std::string tz = "")
```

Comment:
When `tz` is left blank, you end up with a `TimestampType` without timezone awareness (i.e. timezone-naive), as documented here: https://github.com/apache/arrow/blob/56e270fda7f5647a157acd1e428d9735d6399881/cpp/src/arrow/type.h#L1272-L1274 Most places specify two different constructors for it, but using the timezone-aware constructor with an empty string for `tz` is documented to yield a timezone-naive timestamp, so defaulting `tz` to an empty string yields the same thing.

Reply:
I agree that the default behavior here would be to assume a naive timestamp.  I just don't agree that it is a sane default.  The user is explicitly creating a literal so they should know which of the two they want and I don't know how we can possibly make that decision for them (e.g. a naive timestamp is no more likely to be the right answer than an aware timestamp).  So I think it would be better to leave off the default and force the user to tell us what they have.

However, I agree with you that it seems my opinion is against precedent so I'll drop it.